src/domain/sales-channels/tables/product.tsx

```diff
@@ -0,0 +1,462 @@
+
+/* ****************************************** */
+/* ************** TABLE CONFIG ************** */
+/* ****************************************** */
+
+const DEFAULT_PAGE_SIZE = 12
+
+/**
+ * Default filtering config for querying products endpoint.
+ */
+const defaultQueryProps = {
+  additionalFilters: {
+    expand: "collection,sales_channels",
+    fields: "id,title,type,thumbnail,status",
+  },
+  limit: DEFAULT_PAGE_SIZE,
+  offset: 0,
+}
+
+/* ******************************************** */
+/* ************** PRODUCTS TABLE ************** */
+/* ******************************************** */
+
+type ProductTableProps = {
+  isAddTable: boolean
+  count: number
+  products: Product[]
+  setSelectedRowIds: (ids: string[]) => void
+  selectedRowIds: string[]
+  removeProductFromSalesChannel: (ids: string[]) => void
+}
+
+/**
+ * Renders a table of sales channel products.
+ */
+function ProductTable(props: ProductTableProps) {
+  const {
+    productFilters: {
+      setTab,
+      saveTab,
+      removeTab,
+      availableTabs: filterTabs,
+      activeFilterTab,
+      setFilters,
+      reset,
+      filters,
+    },
+    paginate,
+    setQuery: setFreeText,
+    queryObject,
+
+    // CONTAINER props
+    isAddTable,
+    count,
+    products,
+    setSelectedRowIds,
+    removeProductFromSalesChannel,
+  } = props
+
+  const offs = parseInt(queryObject.offset) || 0
+  const limit = parseInt(queryObject.limit)
+
+  const [query, setQuery] = useState(queryObject.query)
+  const [numPages, setNumPages] = useState(0)
+
+  const clearFilters = () => {
+    reset()
+    setQuery("")
+  }
+
+  useEffect(() => {
+    if (typeof count !== "undefined") {
+      const controlledPageCount = Math.ceil(count / limit)
+      setNumPages(controlledPageCount)
+    }
+  }, [count])
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    gotoPage,
+    canPreviousPage,
+    canNextPage,
+    pageCount,
+    nextPage,
+    previousPage,
+    // Get the state from the instance
+    state: { pageIndex, pageSize, ...state },
+  } = useTable(
+    {
+      columns: SALES_CHANNEL_PRODUCTS_TABLE_COLUMNS,
+      data: products || [],
+      manualPagination: true,
+      initialState: {
+        pageIndex: Math.floor(offs / limit),
+        pageSize: limit,
+      },
+      pageCount: numPages,
+      autoResetPage: false,
+      autoResetSelectedRows: false,
+      getRowId: (row) => row.id,
+    },
+    usePagination,
+    useRowSelect
+  )
+
+  useEffect(() => {
+    setSelectedRowIds(Object.keys(state.selectedRowIds))
+  }, [Object.keys(state.selectedRowIds).length])
+
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
+      if (query) {
+        setFreeText(query)
+        gotoPage(0)
+      } else {
+        if (typeof query !== "undefined") {
+          // if we delete query string, we reset the table view
+          reset()
+        }
```

Comment:
Issue: deleting the entire query string doesn't reset the table view

Reply:
This also causes issues if you search for a non-existing product in the detail view it shows the "just-created" screen even if the sales channel has products. We should consider showing an empty table in the case that the sales channel has products but the user searched for a product not on the list: 
![image](https://user-images.githubusercontent.com/88927411/179729825-435abac9-ad76-4ead-bcf0-543b42a0ce25.png)